include/aspect/melt.h

```diff
@@ -271,27 +274,20 @@ namespace aspect
   }
 
 
-  /**
-   * Class that contains all runtime parameters and other helper functions
-   * related to melt transport. A global instance can be retrieved with
-   * SimulatorAccess<dim>::get_melt_handler(), but keep in mind that it only
-   * exists if parameters.include_melt_transport is true.
-   */
-  template <int dim>
-  class MeltHandler: public SimulatorAccess<dim>
+  namespace Melt
   {
-    public:
-      MeltHandler(ParameterHandler &prm);
-
+    template <int dim>
+    struct Parameters
```

Comment:
This struct should only contain the input or other parameters for MeltHandler, not internal variables. In particular `boundary_fluid_pressure`, `is_melt_cell_vector`, and `current_constraints` should be private member variables of the class and not in this struct. If you need them outside, make `get_` functions for them.